### PR TITLE
Incorrect test causing fatal error

### PR DESCRIPTION
**Pull request by zcwilt**

### CODE REVIEW

**Comment by mc12345678** [comment]
Doesn't this "invalidate" the solution of: https://github.com/zencart/zencart/pull/2984#discussion_r369643046?

Is the issue that there is not a test for existence before the is_object test?

This change will if I understand correctly not execute the object method identified here: https://github.com/zencart/zencart/pull/2984/files#diff-39234f367c971f12fac815675944ea4fR201-R203

Or are objects such as `$messageStack` becoming session variables?

**Comment by lat9** [comment]
What was the motivation for the instantiation of the `$messageStack` before the session's and languages' initialization?  Shouldn't messages posted to the stack be based on the currently-active language?

File: includes/classes/InitSystem.php [hunk 1]
```diff
@@ -121,7 +121,7 @@ protected function processAutoTypeObjectMethod($entry)
             $this->actionList[] = ['type' => 'sessionObjectMethod', 'object' => $objectName, 'method' => $methodName];
             $this->debugList[] = 'run session object method - ' . $objectName . ' => ' . $methodName;
         }
-        if (!is_object($_SESSION[$objectName])) {
+        if (is_object($_SESSION[$objectName])) {
```
Comment (drbyte) [inline includes/classes/InitSystem.php]: @zcwilt how does this differ from line 120 above?

**Comment by zcwilt** [comment]
Yes, the fix is incorrect, even though it did `fix' the fatal error.
The problem is the tests themselves will never work correctly. 
This is because the parsing of the autoloaders happens in two phases. 
In phase 1 the autoloaders are reduced to a set of actions in the InitSystem class that are then actually actioned within the autoload_func.php script.
Therefore the $_SESSION array is not actually populated within the InitSystem class, so testing for it there will never work. 
The test should be done within autoload_func.php.
Preparing PR now.

**Comment by mc12345678** [comment]
> What was the motivation for the instantiation of the $messageStack before the session's and languages' initialization? Shouldn't messages posted to the stack be based on the currently-active language?

To offer the ability to push information to the `$messageStack` at a point before all of the information/data necessary to display the message in the session's language has been defined.  At the (brief) moment it would likely be English only; however, I have an idea on how to support incorporating the session's language and will prepare a PR to support.

**Comment by drbyte** [comment]
> What was the motivation for the instantiation of the `$messageStack` before the session's and languages' initialization? Shouldn't messages posted to the stack be based on the currently-active language?

It came from this issue https://github.com/zencart/zencart/issues/2979 by @torvista

**Comment by lat9** [comment]
Just because `$messageStack` is instantiated after the load of the `admin_access` functions doesn't mean that one of those functions doesn't have the message stack available ... it's a _runtime_ thing.

Having not followed the above-posted issue, I'll say that I believe that whatever 'process' is adding a menu item during the auto-load should 'take responsibility' for the to-admin messaging, rather than overloading the admin-access functions.

**Pull request merged**